java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java

```diff
@@ -958,19 +981,21 @@ public SargApplier(SearchArgument sarg,
       filterColumns = mapSargColumnsToOrcInternalColIdx(sargLeaves,
                                                         evolution);
       this.rowIndexStride = rowIndexStride;
+      this.evolution = evolution;
+      exceptionCount = new long[sargLeaves.size()];
+      this.useUTCTimestamp = useUTCTimestamp;
+    }
+
+    public void setRowIndexCols(boolean[] rowIndexCols) {
```

Comment:
Is this change related? Shall we move it a separate ticket if thats the case?

Reply:
It is related. I see that the rowIndexCols are not used within the SArgApplier. Additionally now the RowIndexCols is decided both by the SArgs and the Filters so I wanted to pull this out of the SArgApplier into RecordReaderImpl where it is used.